packages/framework/client-logger/fluid-telemetry/src/test/containerTelemetryEndToEnd.spec.ts

```diff
@@ -0,0 +1,251 @@
+					// (this is an example string)
+					"InstrumentationKey=abcdefgh-ijkl-mnop-qrst-uvwxyz6ffd9c;IngestionEndpoint=https://westus2-2.in.applicationinsights.azure.com/;LiveEndpoint=https://westus2.livediagnostics.monitor.azure.com/",
+			},
+		});
+		appInsightsTrackEventSpy = spy(appInsightsClient, "trackEvent");
+	});
+
+	beforeEach(() => {
+		// We need to reset the Application Insights client spy for each test so the trackEvent calls
+		// from the last test don't bleed into the next one.
+		appInsightsTrackEventSpy.resetHistory();
+	});
+
+	it("IFluid container's 'connected' system event produces expected ContainerConnectedTelemetry using AppInsightsTelemetryConsumer", async () => {
+		const { container } = await tinyliciousClient.createContainer(schema);
+
+		const containerId = await container.attach();
+		startTelemetry({
+			container,
+			containerId,
+			consumers: [new AppInsightsTelemetryConsumer(appInsightsClient)],
+		});
+
+		// We Capture and analyze telemetry after the container connected event
+		let didTestPass = false;
+		container.on("connected", () => {
+			// Obtain the calls made to the appInsights trackEvent method
+			const actualEmittedAppInsightsTelemetry: AppInsights.IEventTelemetry[] =
+				appInsightsTrackEventSpy
+					.getCalls()
+					.map((spyCall) => spyCall.args[0] as AppInsights.IEventTelemetry)
+					.filter(
+						(telemetry) => telemetry.name === ContainerTelemetryEventNames.CONNECTED,
+					);
+
+			if (actualEmittedAppInsightsTelemetry.length === 0) {
+				assert.fail(
+					"Expected AppInsights.trackEvent() to be called alteast once with expected container telemetry but was not.",
+				);
+			}
+			const actualAppInsightsTelemetry = actualEmittedAppInsightsTelemetry[0];
+			const actualContainerTelemetry =
+				actualAppInsightsTelemetry.properties as IContainerTelemetry;
+
+			const expectedAppInsightsTelemetry: AppInsights.IEventTelemetry = {
+				name: ContainerTelemetryEventNames.CONNECTED,
+				properties: {
+					eventName: ContainerTelemetryEventNames.CONNECTED,
+					containerId,
+					containerInstanceId: actualContainerTelemetry.containerInstanceId,
+				} satisfies ContainerConnectedTelemetry,
+			};
+
+			expect(expectedAppInsightsTelemetry).to.deep.equal(actualAppInsightsTelemetry);
+			// We won't know what the container containerInstanceId will be but we can still check that it is defined.
+			expect(actualContainerTelemetry.containerInstanceId)
+				.to.be.a("string")
+				.with.length.above(0);
+
+			// This will enable the while loop at the end of this test to finally complete,
+			// We  want the test to complete as soon as we recieve the container event and complete our testing
+			didTestPass = true;
+		});
+
+		// We don't know exactly when the given container events will fire and we can't await specific events so we have to
+		// have a while loop that keeps the test alive until the container notifies us with its event handlers.
+		while (!didTestPass) {
```

Comment:
Nit: instead of implementing this as a while loop, we could just return a promise that resolves when our listener gets called. The test will timeout and fail if that doesn't happen within the test time limit.